src/lib/authorizationURL.ts

```diff
@@ -29,6 +30,14 @@ function getAuthorizationURL(config: OAuthAgentConfiguration): AuthorizationRequ
         "&code_challenge=" + generateHash(codeVerifier) +
         "&code_challenge_method=S256"
 
+    if (options && options.extraParams) {
+        options.extraParams.forEach((p) => {
+            if (p.key && p.value) {
+                authorizationRequestUrl += `&${p.key}=${encodeURIComponent(p.value)}`
```

Comment:
I think that the key should also be encoded as it contains input from the user (or which can be manipulated by the user/xss/etc.). Maybe we should be even more strict and sanitize that input? E.g. make sure that neither the key nor the value contains unencoded `&` or `=` characters.